File: app/src/main/java/com/zwstudio/logicpuzzlesandroid/common/data/LevelProgress.java
package com.zwstudio.logicpuzzlesandroid.common.data;

import com.j256.ormlite.field.DatabaseField;

public class LevelProgress implements java.io.Serializable {
    @DatabaseField(generatedId = true)
    private int ID;
    @DatabaseField
    public String gameID;
    @DatabaseField
    public String levelID;
    @DatabaseField
    public int moveIndex;
}

File: app/src/main/java/com/zwstudio/logicpuzzlesandroid/common/data/LevelProgress.kt
package com.zwstudio.logicpuzzlesandroid.common.data

import io.realm.RealmObject
import io.realm.annotations.PrimaryKey
import java.util.*

open class LevelProgress : RealmObject() {
    @PrimaryKey
    var id = UUID.randomUUID().toString()
    var gameID: String? = null
    var levelID: String? = null
    var moveIndex = 0
}